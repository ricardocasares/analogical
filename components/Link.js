// internal
import { Link } from "../routes";

export default props => (
  <Link {...props}>
    <a>{props.children}</a>
  </Link>
);
