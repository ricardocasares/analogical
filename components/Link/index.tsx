import Link, { LinkProps } from "next/link";
import React, { Children, FunctionComponent } from "react";
import { withRouter, WithRouterProps } from "next/router";

interface ActiveLink extends LinkProps, WithRouterProps {
  partial?: boolean;
}

export const match = (exact: boolean) => (href: string, pathname: string) => {
  if (exact) return pathname.includes(href);
  return href === pathname;
};

const ActiveLink: FunctionComponent<ActiveLink> = ({
  router,
  children,
  partial = false,
  ...props
}) => {
  const child = Children.only(children);

  let className = null;
  const test = match(partial);

  if (router && test(props.href as string, router.pathname)) {
    className = "active";
  }

  return <Link {...props}>{React.cloneElement(child, { className })}</Link>;
};

export default withRouter(ActiveLink);
