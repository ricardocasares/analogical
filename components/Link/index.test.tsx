import React from "react";
import renderer from "react-test-renderer";
import serializer from "jest-emotion";
import Link from "@app/components/Link";

expect.addSnapshotSerializer(serializer);

test("renders with correct styles", () => {
  expect(
    renderer
      .create(
        <Link href="/">
          <a>hello world</a>
        </Link>
      )
      .toJSON()
  ).toMatchSnapshot();
});

test("renders partial with correct styles", () => {
  expect(
    renderer
      .create(
        <Link href="/" partial>
          <a>hello world</a>
        </Link>
      )
      .toJSON()
  ).toMatchSnapshot();
});
