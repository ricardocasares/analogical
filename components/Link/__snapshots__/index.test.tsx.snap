// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders partial with correct styles 1`] = `
<a
  className={null}
  href="/"
  onClick={[Function]}
>
  hello world
</a>
`;

exports[`renders with correct styles 1`] = `
<a
  className={null}
  href="/"
  onClick={[Function]}
>
  hello world
</a>
`;
